src/components/Landing.js

```diff
@@ -237,109 +238,124 @@ export default class Landing extends Component {
-        summary[overviewSectionKey+"_graph"] = graph_data;
-      } else {
-        sections.forEach(item => {
-          if (summary[item.section_key] && summary[item.section_key][item.subSection_key]) {
-            //console.log("section data? ", summary[item.section_key][item.subSection_key])
-            graph_data = [...graph_data, ...summary[item.section_key][item.subSection_key]];
-          }
-        });
-        let prevObj = null;
-        let o = {};
-        const PLACEHOLDER_FIELD_NAME = "placeholder";
-        const compareDate = (a, b) => {
-          let calcA = (new Date(a["Start"])).getTime();
-          let calcB = (new Date(b["Start"])).getTime();
-          if (calcA > calcB) return 1;
-          if (calcB > calcA) return -1;
-          return 0;
+    if (!(graphConfig && graphConfig.summaryDataSource)) {
+      return;
+    }
+    //get the data from summary data
+    let sections = graphConfig.summaryDataSource;
+    let graph_data = [];
+    if (getEnv(graphConfig.demoConfigKey)) {
+      graph_data = graphConfig.demoData;
+      summary[overviewSectionKey+"_graph"] = graph_data;
+    } else {
+      sections.forEach(item => {
+        if (summary[item.section_key] && summary[item.section_key][item.subSection_key]) {
+          graph_data = [...graph_data, ...summary[item.section_key][item.subSection_key]];
         }
-        graph_data = graph_data.sort(compareDate);
-        /*
-         * processing graph data
-         */
-        graph_data.forEach(function(item, index) {
-          /* make sure dates are in consistent format */
-          let startDate = item[graphConfig.startDateField];
-          if (startDate.indexOf("T") > 0) {
-            startDate = startDate.substring(0, startDate.indexOf("T"));
-          }
-          let endDate = item[graphConfig.endDateField];
-          if (endDate.indexOf("T") > 0) {
-            endDate = endDate.substring(0, endDate.indexOf("T"));
-          }
-          //baseline point
-          if (index === 0 && item[graphConfig.mmeField] !== 0) {
-            o = {};
-            o[graphConfig.graphDateField]=startDate;
-            o[graphConfig.mmeField] = 0;
-            o[PLACEHOLDER_FIELD_NAME] = true;
-            formattedGraphData.unshift(o);
+      });
+      const [startDateFieldName, endDateFieldName, MMEValueFieldName, graphDateFieldName] = [graphConfig.startDateField, graphConfig.endDateField, graphConfig.mmeField, graphConfig.graphDateField];
+      const DELIMITER_FIELD_NAME = "delimiter";
+      const PLACEHOLDER_FIELD_NAME = "placeholder";
+
+      //sort data by start date
+      graph_data = graph_data.filter(function(item) {
+        return item[startDateFieldName] && item[endDateFieldName];
+      }).sort(function(a, b) {
+        return dateCompare(a[startDateFieldName], b[startDateFieldName]);
+      });
+
+      let dataPoints = [];
+      let prevObj = null, nextObj = null;
+      graph_data.forEach(function(item, index) {
```

Comment:
In larger loops, you may want to rename your loop variables (`item`) to something more obvious so it's easier to keep track of what's being iterated over